packages/ra-core/src/controller/ListController.tsx

```diff
@@ -310,9 +395,11 @@ export class ListController extends Component {
             onSelect: this.handleSelect,
             onToggleItem: this.handleToggleItem,
             onUnselectItems: this.handleUnselectItems,
-            page: parseInt(query.page || 1, 10),
-            perPage: parseInt(query.perPage, 10),
-            refresh: this.refresh,
+            page:
+                (typeof query.page === 'string'
+                    ? parseInt(query.page, 10)
+                    : query.page) || 1,
+            perPage: parseInt(query.perPage.toString(), 10),
```

Comment:
please use the same test as in the previous line. I prefer not casting to string to get a number in the end